src/Home.js

```diff
@@ -105,16 +54,84 @@ export default function Home() {
         </Pressable>
       </View>
     </View>
-  </Modal>)
+  </Modal>);
+}
+
+const TodoList = ({ todos }) => {
+
+  async function deleteTodo(todo) {
+    try {
+      await DataStore.delete(todo);
+    } catch (e) {
+      console.log(`Delete failed: ${e}`);
+    }
+  }
+
+  async function setComplete(updateValue, todo) {
+    await DataStore.save(
+      Todo.copyOf(todo, updated => {
+        updated.isComplete = updateValue
+      })
+    );
+  }
+
+  const todoItem = ({item}) => (
+    <Pressable style={styles.todoContainer}
+      onLongPress={() => {
+        deleteTodo(item);
+      }}>
+      <View>
+        <Text style={styles.todoHeading}>{item.name}</Text>
+        <Text style={styles.todoDescription}>{item.description}</Text>
+      </View>
+      <Pressable
+        style={[styles.checkbox, item.isComplete && styles.selectedCheckbox]}
+        onPress={() => {
+          setComplete(!item.isComplete, item)
+        }}
+      >
+        <Text style={[styles.checkboxText, item.isComplete && styles.selectedCheckboxText]}>{(item.isComplete) ? "✓" : ""}</Text>
+      </Pressable>
+    </Pressable >
+  )
+
+  return (
+    < FlatList
+      data={todos}
+      renderItem={todoItem}
+      keyExtractor={item => item.id}
+    />)
+};
+
+
+export default function Home() {
+
+  const [todos, setTodos] = useState([]);
+  const [modalVisible, setModalVisible] = useState(false);
+
+  useEffect(() => {
```

Comment:
Looks like this `useEffect` could just be in `TodoList`?

Reply:
Good point. Makes the home component much smaller.